subsys/bluetooth/host/classic/avdtp.c

```diff
@@ -420,13 +422,16 @@ static void avdtp_get_capabilities_rsp(struct bt_avdtp *session, struct net_buf
 	}
 
 	k_work_cancel_delayable(&session->timeout_work);
-	GET_CAP_REQ(session->req)->buf = NULL;
+	avdtp_set_status(req, buf, msg_type);
 
 	if (msg_type == BT_AVDTP_ACCEPT) {
+		/* move the ref of buf to GET_CAP_REQ(req)->buf */
 		GET_CAP_REQ(req)->buf = buf;
+		buf = NULL;
+	} else {
+		GET_CAP_REQ(req)->buf = NULL;
```

Comment:
Aren't you missing a `net_buf_unref(buf);` in this branch? Seems like you are, and then you can move the `buf = NULL;` to be a single common step after both branches.

Reply:
The `buf` comes from the l2cap through the L2CAP `recv` callback `bt_avdtp_l2cap_recv`. avdtp doesn't ref the `buf`, so I think avdtp doesn't need to unref it too.
I change the `req->func(req)` callback function to pass the `buf` as one parameter not give the `buf` to another variable, so I think the `ref/unref` is not needed. please see the last commit in the pr.

Reply:
Then this function doesn't own the ref, and if you don't own the ref then you can't move it either. In that case the only thing the function needs to do is a `ref()` as part of assigning to the `req`.

Reply:
I have changed the codes, there is no assigning `buf` to the `req` now. I pass the `buf` as parameter that has similar behavior like the L2CAP callback.